Uchu.StandardScripts/General/StatPickups.cs

```diff
@@ -8,63 +10,34 @@ public class StatPickups : NativeScript
+                {Lot.TwoArmor, 909},
+                {Lot.ThreeArmor, 910},
+                {Lot.FiveArmor, 911},
+                {Lot.TenArmor, 912},
+            };
             Listen(Zone.OnPlayerLoad, player =>
             {
                 Listen(player.OnLootPickup, lot =>
                 {
+                    if (!pickups.ContainsKey(lot)) return Task.CompletedTask;
                     var stats = player.GetComponent<DestroyableComponent>();
-                    
-                    switch (lot)
-                    {
-                        case Lot.Imagination:
-                            stats.Imagination += 1;
-                            break;
-                        case Lot.TwoImagination:
-                            stats.Imagination += 2;
-                            break;
-                        case Lot.ThreeImagination:
-                            stats.Imagination += 3;
-                            break;
-                        case Lot.FiveImagination:
-                            stats.Imagination += 5;
-                            break;
-                        case Lot.TenImagination:
-                            stats.Imagination += 10;
-                            break;
-                        case Lot.Health:
-                            stats.Health += 1;
-                            break;
-                        case Lot.TwoHealth:
-                            stats.Health += 2;
-                            break;
-                        case Lot.ThreeHealth:
-                            stats.Health += 3;
-                            break;
-                        case Lot.FiveHealth:
-                            stats.Health += 5;
-                            break;
-                        case Lot.TenHealth:
-                            stats.Health += 10;
-                            break;
-                        case Lot.Armor:
-                            stats.Armor += 1;
-                            break;
-                        case Lot.TwoArmor:
-                            stats.Armor += 2;
-                            break;
-                        case Lot.ThreeArmor:
-                            stats.Armor += 3;
-                            break;
-                        case Lot.FiveArmor:
-                            stats.Armor += 5;
-                            break;
-                        case Lot.TenArmor:
-                            stats.Armor += 10;
-                            break;
-                        default:
-                            return Task.CompletedTask;
-                    }
-                    
+                    var skill = player.GetComponent<SkillComponent>();
+                    var missionInventoryComponent = player.GetComponent<MissionInventoryComponent>();
+                    skill.CalculateSkillAsync(pickups[lot], player);
+                    missionInventoryComponent.StatPickupsAsync(pickups[lot]);
```

Comment:
Should we do the StatPickup mission task updates here or in the same place as part of wherever we update skill mission tasks? They are practically the same. Alternatively, `StatPickupsAsync` could be merged with `UseSkillAsync`.

Reply:
Given how it's triggered by a Skill anyway, I'll merge it with UseSkillAsync

Reply:
Do make sure there aren't any items or weapons that also use this skill to restore stats